frontend/catalyst/jax_primitives.py

```diff
@@ -386,6 +409,138 @@ def _print_lowering(jax_ctx: mlir.LoweringRuleContext, *args, string=None, memre
+    with_named_sequence_attr = ir.UnitAttr.get(jax_ctx.module_context.context)
+    module.operation.attributes["transform.with_named_sequence"] = with_named_sequence_attr
+
+    # Insert the transform.named_sequence op into the module to be transformed
+    # Note that InsertionPoint(Block) inserts after the last operation but still inside the block.
+    # Hence we insert at the end, inside the module
+    with ir.InsertionPoint(module.body):
+        named_sequence_op = NamedSequenceOp(
+            sym_name="__transform_main",
+            function_type=functype_attr,
+        )
+
+        # transform.named_sequence op is the "main function" of the transform dialect
+        # and thus needs an entry block (which also should be its only block)
+        # The argument of the block is the target function to run the transform sequence on
+        bb = ir.Block.create_at_start(named_sequence_op.body, arg_types=[transform_func_type])
+
+        # The transform.named_sequence needs a terminator called "transform.yield"
+        with ir.InsertionPoint(bb):
+            transform_yield_op = TransformYieldOp(operands_=[])  # pylint: disable=unused-variable
+
+    return named_sequence_op.results
+
+
+#
+# apply_registered_pass
+#
+@apply_registered_pass_p.def_abstract_eval
+def _apply_registered_pass_abstract_eval(*args, pass_name, options=None):
+    return AbstractTransformFunc()
+
+
+@apply_registered_pass_p.def_impl
+def _apply_registered_pass_def_impl(*args, pass_name, options=None):  # pragma: no cover
+    raise NotImplementedError()
+
+
+def _apply_registered_pass_lowering(
+    jax_ctx: mlir.LoweringRuleContext, *args, pass_name, options=None
+):
+    module = jax_ctx.module_context.module
+    named_sequence_op = None
+    for op in reversed(module.body.operations):
+        # transform.named_sequence usually is at the end of the module, so look for it from the end
+        if op.operation.name == "transform.named_sequence":
+            named_sequence_op = op.operation
+            break
+    if named_sequence_op is None:
+        raise RuntimeError(
+            """
+            transform.apply_registered_pass must be placed in a transform.named_sequence, 
+            but none exist in the module.
+            """
+        )
+
+    # If there already is a apply_registered_pass,
+    # insert after the last pass in the existing pass sequence.
+    # Note that ir.InsertionPoint(op) sets the insertion point to immediately BEFORE the op
+    named_sequence_op_block = named_sequence_op.regions[0].blocks[0]
+    first_op_in_block = named_sequence_op_block.operations[0].operation
+
+    assert first_op_in_block.name in (
+        "transform.apply_registered_pass",
+        "transform.yield",
+    ), """
+            Unexpected operation in transform.named_sequence! 
+            Only transform.apply_registered_pass and transform.yield are allowed.
+        """
+
+    if first_op_in_block.name == "transform.apply_registered_pass":
```

Comment:
Here again the codecov messages probably indicate we don't have a pytest with multiple passes being registered?

Reply:
A good pytest would be a qnode inside and outside qjit when being cancel-inversed gives the same result. I think this will hit this line since these will be two distinct functions and hence two apply_pass primitives. 

It's also a nice test to have on its own too.

Reply:
done